extensions/ONNX/Onnx/OnnxTextGenerator.cs

```diff
@@ -0,0 +1,138 @@
+/// See https://github.com/microsoft/onnxruntime-genai
+/// </summary>
+[Experimental("KMEXP01")]
+public sealed class OnnxTextGenerator : ITextGenerator, IDisposable
+{
+    private readonly Model? _model = default;
+    private readonly Tokenizer? _tokenizer = default;
+    private readonly ILogger<OnnxTextGenerator> _log;
+    private readonly ITextTokenizer _textTokenizer;
+
+    /// <summary>
+    /// Create a new instance
+    /// </summary>
+    /// <param name="config">Configuration settings</param>
+    /// <param name="textTokenizer">Text Tokenizer</param>
+    /// <param name="loggerFactory">Application Logger instance</param>
+    public OnnxTextGenerator(
+        OnnxConfig config,
+        ITextTokenizer? textTokenizer = null,
+        ILoggerFactory? loggerFactory = null)
+    {
+        this._log = (loggerFactory ?? DefaultLogger.Factory).CreateLogger<OnnxTextGenerator>();
+        if (textTokenizer == null)
+        {
+            this._log.LogWarning(
+                "Tokenizer not specified, will use {0}. The token count might be incorrect, causing unexpected errors",
+                nameof(GPT4oTokenizer));
+            textTokenizer = new GPT4oTokenizer();
+        }
+
+        config.Validate();
+        this.MaxTokenTotal = (int)config.MaxLength;
+        this.MinLength = config.MinLength;
+        this.PastPresentShareBuffer = config.PastPresentShareBuffer;
+        this._model = new Model(config.ModelPath);
+        this._tokenizer = new Tokenizer(this._model);
+        this._textTokenizer = textTokenizer;
+
+        var modelFilename = config.ModelPath.TrimEnd('/').Split('/').Last();
+        this._log.LogDebug("Loading Onnx model: {0}", modelFilename);
+        this._model = new Model(config.ModelPath);
+        this._tokenizer = new Tokenizer(this._model);
+        this._log.LogDebug("Onnx model loaded");
+    }
+
+    /// <inheritdoc/>
+    public int MaxTokenTotal { get; }
+
+    /// <inheritdoc/>
+    public double MinLength { get; }
+
+    /// <inheritdoc/>
+    public bool PastPresentShareBuffer { get; }
+
+    /// <inheritdoc/>
+    public async IAsyncEnumerable<string> GenerateTextAsync(
+        string prompt,
+        TextGenerationOptions? options = null,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        var tokens = this._tokenizer?.Encode(prompt);
+        using var generatorParams = new GeneratorParams(this._model);
+
+        generatorParams.SetSearchOption("max_length", (double)this.MaxTokenTotal);
+        generatorParams.SetSearchOption("min_length", this.MinLength);
+        generatorParams.SetSearchOption("past_present_share_buffer", this.PastPresentShareBuffer);
+
+        if (options != null)
+        {
+            generatorParams.SetSearchOption("temperature", options.Temperature);
+        }
+
+        await Task.Run(() => generatorParams.SetInputSequences(tokens), cancellationToken).ConfigureAwait(true);
```

Comment:
`generatorParams` could be disposed outside

why the need to run a task this way?

Reply:
Using a Task.Run to remove the compiler warning about the method not being async without an await. 

Removing async from the method entirely throw [Error CS8403](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs8403?f1url=%3FappId%3Droslyn%26k%3Dk(CS8403))